electron/legendary_utils/library.ts

```diff
@@ -21,45 +21,33 @@ const execAsync = promisify(exec)
 const statAsync = promisify(stat)
 
 export async function getLegendaryGames() {
-  const { stdout, stderr } = await execAsync(
-    `${legendaryBin} list-games --json`
-  )
-  if (stdout) {
-    const results = JSON.parse(stdout)
-    // NEED DOUBLE CHECK IF SOMETHINGS INSIDE IS NEEDED OR ADD MORE STUFF IT REDUCE SIZE OF THE FINAL FILE
-    // There is maybe a better way to do that tried with filter but still not work
-    results.map((res: any) => {
-      delete res.asset_info
-      delete res.metadata.creationDate
-      delete res.metadata.developerId
-      delete res.metadata.dlcItemList
-      delete res.metadata.endOfSupport
-      delete res.metadata.entitlementName
-      delete res.metadata.entitlementType
-      delete res.metadata.lastModifiedDate
-      delete res.metadata.releaseInfo
-      res.metadata.keyImages.map((k: any) => {
-        delete k.height
-        delete k.md5
-        delete k.size
-        delete k.uploadedDate
-        delete k.width
-      })
-    })
-    const json = JSON.stringify(results)
-    return writeFileSync(`${heroicFolder}library.json`, json)
-  } else if (stderr) {
-    // return a file to avoid infinite loop
-    return writeFileSync(
-      `${heroicFolder}library.json`,
-      '{"result":"noGamesFound"}'
-    )
-  } else {
-    // return a file to avoid infinite loop
-    return writeFileSync(
-      `${heroicFolder}library.json`,
-      '{"result":"noGamesFound"}'
-    )
+  const {stdout, stderr} = await execAsync(`${legendaryBin} list-games --json`)
```

Comment:
The issue here wasn't the size of the file but the stdout of the list-games command. IF you have too many games, the output will be really big and the process will complain so this solution will still have the same issue unfortunatelly.
I think the current one of reading the metadata directory is not the ideal but at least it has no issues with the stdout.
Maybe there is a way to surpass this limit but I'm not sure if it is possible to do it on the node service or it needs to be a configuration on Linux.
This was also the reason why I removed the output of the instalation after finishing it, because it was giving me stdout errors.

Reply:
If stdout is a string type that gets too long, you could get the command output to be saved to a temporary .json file, then parse that file as an object directly to avoid it being a string at any point until you do JSON.stringify()

Reply:
I thought about doing that but I guess it will have too much overhead. Because this command is called everytime you refresh the library so can be slow. Like on first login, for instance. So on slower machines could take too long, I'm not sure about it.